cmd/kubelet/app/server.go

```diff
@@ -1294,7 +1294,7 @@ func RunDockershim(f *options.KubeletFlags, c *kubeletconfiginternal.KubeletConf
 	}
 
 	// Standalone dockershim will always start the local streaming server.
-	ds, err := dockershim.NewDockerService(dockerClientConfig, r.PodSandboxImage, streamingConfig, &pluginSettings,
+	ds, err := dockershim.NewDockerService(dockerClientConfig, r.PodSandboxImage, r.PodSandboxSeccomp, streamingConfig, &pluginSettings,
```

Comment:
What's the plan for containerd / cri-o? I assume the a similar change needs to be merged into those repos? Or can we control the infra container for remote runtimes from within the Kubelet?

Reply:
We can do it within the kubelet. I will tag you on the line of code I am currently doing that.

Reply:
We can only partially do this from the Kubelet. For sandboxes, we can only use seccomp paths using the current syntax: `runtime/name` or `localhost/name`. We cannot provide the full seccomp json. To use the custom profile created on this PR for non dockershim, we would have to ensure the profile is written on disk and refer to it with the `localhost` prefix. However, that would still not work, because we would need to pass on the `no-new-privilege` flag, which I believe we currently can only do for the "normal containers".

For those reasons, the Alpha version defaults the sandbox container seccomp profile to `runtime/default` when using non dockershim runtimes. The end result is the same, users will be able to interchance their container and pod level profiles with least privileges. However, sandboxes running in dockershim will operate at tighter security profiles.

The longer term of this could be a new built-in profile type `kubernetes/sandbox` plus changes on the CRI API for the no-new-priv flag.

Reply:
The kubelet doesn't actually have any knowledge of the infra container when interacting with the CRI directly - the sandbox could be anything. So, the fact that containerd & cri-o are running the pause container is really an implementation detail of those runtimes, and I think this change needs to be baked into those runtimes.

Given that, I think the best way to handle this would be to add this profile as a library alongside the pause source: https://github.com/kubernetes/kubernetes/tree/master/build/pause, and then import the profile to dockershim, as well as containerd & cri-o. Ideally, profile could just be a json file in the paues directory, but I'm not sure what the best way to turn that into a go importable library are.

Reply:
@tallclair you are absolutely right. I will take this into consideration and see how we can bridge that gap before making seccomp GA. I have a few ideas and will explore them on tomorrow's call. I will forward you the details, in case you are available then.